src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePump.cs

```diff
@@ -0,0 +1,101 @@
+            var messageReceiver = CreateMessageReceiver();
+            Logger.LogInformation("Starting message pump on entity path {EntityPath} in namespace {Namespace}", EntityPath, Namespace);
+
+            // TODO: Message pump options to not delete for example
+            var messageHandlerOptions = new MessageHandlerOptions(HandleReceivedException);
+            messageReceiver.RegisterMessageHandler(HandleMessage, messageHandlerOptions);
+            Logger.LogInformation("Message pump started");
+
+            await UntilCancelledAsync(stoppingToken);
+
+            Logger.LogInformation("Closing message pump");
+            await messageReceiver.CloseAsync();
+            Logger.LogInformation("Message pump closed : {Time}", DateTimeOffset.UtcNow);
+        }
+
+        private MessageReceiver CreateMessageReceiver()
+        {
+            var connectionString = Configuration.GetValue<string>("ARCUS_SERVICEBUS_QUEUE_CONNECTIONSTRING");
+
+            var serviceBusConnectionStringBuilder = new ServiceBusConnectionStringBuilder(connectionString);
+
+            var messageReceiver = new MessageReceiver(serviceBusConnectionStringBuilder);
+            
+            EntityPath = serviceBusConnectionStringBuilder.EntityPath;
+            Namespace = messageReceiver.ServiceBusConnection.Endpoint.Host;
+
+            return messageReceiver;
+        }
+
+        private async Task HandleReceivedException(ExceptionReceivedEventArgs exceptionEvent)
+        {
+            await HandleReceiveExceptionAsync(exceptionEvent.Exception);
+        }
+
+        private async Task HandleMessage(Message message, CancellationToken cancellationToken)
+        {
+            var correlationInfo = new MessageCorrelationInfo(message.GetTransactionId(), message.CorrelationId);
+            var messageContext =
+                new AzureServiceBusMessageContext(message.MessageId, message.SystemProperties, message.UserProperties);
+
+            Logger.LogInformation("Received message '{MessageId}' (Transaction: {TransactionId}, Operation: {OperationId}, Cycle: {CycleId})",
+                messageContext.MessageId, correlationInfo.TransactionId, correlationInfo.OperationId,
+                correlationInfo.CycleId);
+            
+            // Deserialize the message
+            var typedMessageBody = DeserializeJsonMessageBody(message.Body, messageContext);
+
+            // Process the message
+            // Note - We are not checking for exceptions here as the pump wil handle those and call our exception handling after which it abandons it
+            await ProcessMessageAsync(typedMessageBody, messageContext, correlationInfo, cancellationToken);
+            
+            Logger.LogInformation("Message {MessageId} processed", message.MessageId);
+        }
+
+        private static Task UntilCancelledAsync(CancellationToken cancellationToken)
+        {
+            var taskCompletionSource = new TaskCompletionSource<bool>();
```

Comment:
Not sure this is the best way of creating a long-running task.
Either way TaskCompletionSource should be created using TaskCreationOptions.RunContinuationsAsynchronously
another option here could simply be Task.Delay(Timeout.Infinite, cancellationToken)

But I still think we need to discuss options for creating long running tasks that are effectively blocking the thread.

Reply:
Pulling in @fgheysels as I've used his proposal, don't have a preference.

Reply:
Switched to using `await Task.Delay(Timeout.Infinite, cancellationToken);` for now.

Good suggestion @wseye

Reply:
Added check for `IsCancellationRequested` as well